### PR TITLE
Allow only common trigger options in multi-trigger apps

### PR DESCRIPTION
**Pull request by itowlson**
Alternative to #2255 to address #2251

### CODE REVIEW

File: src/commands/up.rs [hunk 1]
```diff
@@ -410,6 +423,57 @@ impl UpCommand {
             }
         }
     }
+
+    fn allow_only_common_flags(&self) -> anyhow::Result<()> {
+        allow_only_common_flags(&self.trigger_args)
+    }
+}
+
+const COMMON_FLAGS: &[&str] = &[
```
Comment (rylev) [inline src/commands/up.rs:432]: Can we comment that these flags are common *to all well-known trigger types*.

File: src/commands/up.rs [hunk 2]
```diff
@@ -410,6 +423,57 @@ impl UpCommand {
...
+    "--runtime-config-file",
+    "--state-dir",
+    "--key-value",
+    "--sqlite",
+    "--help-args-only",
+];
+
+fn allow_only_common_flags(args: &[OsString]) -> anyhow::Result<()> {
+    for i in 0..(args.len()) {
+        let Some(value) = args[i].to_str() else {
```
Comment (rylev) [inline src/commands/up.rs:453]:
```suggestion
    for (i, arg) in args.iter().enumerate() {
        let Some(value) = arg.to_str() else {
```

File: src/commands/up.rs [hunk 3]
```diff
@@ -410,6 +423,57 @@ impl UpCommand {
...
+            // Err on the side of forgiveness
+            continue;
+        };
+        if value.starts_with('-') {
+            // Flag candidate. Is it allowed?
+            if !is_common_flag(value) {
```
Comment (rylev) [inline src/commands/up.rs:459]: Nit: I would personally inline this function.

**Review by rylev — APPROVED** [review]
This looks good to me!

**Comment by itowlson** [comment]
@rylev Thanks for the review - all good points.  Unfortunately, as we want to land this for 2.2, I'm going to have to punt them to a tidy-up PR - sorry.  Also, as I'm already looking at Option 3 (and have it working on the happy path), which would enable trigger-specific flags, tidyings to this may become moot, but we'll see how it goes.

**Pull request merged**